server/getCredentials.ts

```diff
@@ -1,17 +1,26 @@
 "use server";
 
 import { Credentials } from "@/app/types";
- const validLoginDetails: Credentials[] = [
-  {
-    username: "FakeUser1",
-    password: "Password1",
-  },
-  {
-    username: "FakeUser2",
-    password: "Password2",
-  },
-];
 
-export async function getCredentials(): Promise<Credentials[]> {
-  return validLoginDetails;
+export async function credentialsAreValid(
+  inputUsername: string,
+  inputPassword: string
+) {
+  const validLoginDetails: Credentials[] = [
+    {
+      username: "FakeUser1",
+      password: "Password1",
+    },
+    {
+      username: "FakeUser2",
+      password: "Password2",
+    },
+  ];
+  const elementWithMatchingUsername = validLoginDetails.find((loginDetails) => {
+    return loginDetails.username === inputUsername;
+  });
+  if (!elementWithMatchingUsername) {
+    return;
+  }
+  return elementWithMatchingUsername.password === inputPassword;
```

Comment:
there's a different array method that would do this more concisely

Reply:
https://github.com/auberginearms/starterApp/pull/1/commits/db49e2a731d23e491395235c649b063cc23da78d